result.go

```diff
@@ -31,54 +33,80 @@ type Result struct {
 	FullName      string         `json:"fullName,omitempty"`
 	Labels        []Label        `json:"labels,omitempty"`
 }
+type FailureMode string
 
 //Before defines a step
 type Before struct {
 	Name          string         `json:"name,omitempty"`
 	Status        string         `json:"status,omitempty"`
-	StatusDetails *StatusDetails `json:"statusDetails,omitempty"`
+	StatusDetails *statusDetails `json:"statusDetails,omitempty"`
 	Stage         string         `json:"stage,omitempty"`
 	Description   string         `json:"description,omitempty"`
 	Start         int64          `json:"start,omitempty"`
 	Stop          int64          `json:"stop,omitempty"`
-	Steps         []Step         `json:"steps,omitempty"`
+	Steps         []stepObject   `json:"steps,omitempty"`
 	Attachments   []Attachment   `json:"attachments,omitempty"`
 }
 
+type HasSteps interface {
+	GetSteps() []stepObject
+	AddStep(step stepObject)
+}
+
+func (r *result) GetSteps() []stepObject {
+	return r.Steps
+}
+
+func (r *result) AddStep(step stepObject) {
+	r.Steps = append(r.Steps, step)
+}
+
 type Parameter struct {
 	Name  string `json:"name,omitempty"`
 	Value string `json:"value,omitempty"`
 }
 
 var wsd, resultPath string
 
+const (
+	ALLURE_RESULTS_PATH = "ALLURE_RESULTS_PATH"
```

Comment:
please avoid ALL_CAPS_CONST_OR_VAR, and follow go conventions.

```suggestion
	resultsPathEnvKey = "ALLURE_RESULTS_PATH"
```

Reply:
Done.